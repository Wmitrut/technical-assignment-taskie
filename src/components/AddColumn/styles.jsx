import styled from 'styled-components';
import { COLORS } from '../../constants/colors';

export const AddButton = styled.button`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  width: 200px;
  padding: 15px 5px;
  border: none;
  background: ${COLORS.primary};
  border: 1px solid;
  border-color: ${COLORS.secondary};
  box-sizing: border-box;
  border-radius: 10px;
  color: #fff;
  margin-left: 10px;
  transition: 0.3s all;
  cursor: pointer;

  &:hover {
    background: ${COLORS.primaryDark};
  }
`;

export const InputBoard = styled.input`
  display: flex;
  font-family: Inter;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  width: 360px;
  padding: 15px 5px;
  border: none;
  background: ${COLORS.primaryDark};
  border: 1px solid;
  border-color: ${COLORS.secondary};
  box-sizing: border-box;
  border-radius: 10px;
  color: ${COLORS.light};
  margin-left: 10px;
  transition: 0.3s all;
  cursor: pointer;

  &:focus {
    outline: none;
    background: ${COLORS.primary};
  }

  &:hover {
    background: ${COLORS.primary};
  }
`;

export const SaveButton = styled.button`
  display: flex;
  font-family: Inter;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  width: 100px;
  padding: 10px 5px;
  border: none;
  background: ${COLORS.primary};
  border: 1px solid;
  border-color: ${COLORS.secondary};
  box-sizing: border-box;
  border-radius: 10px;
  color: ${COLORS.light};
  transition: 0.3s all;
  cursor: pointer;
  margin-left: 10px;

  &:hover:disabled {
    background: ${COLORS.successDisabled};
    cursor: not-allowed;
  }

  &:hover {
    background: ${COLORS.success};
  }
`;

export const CancelButton = styled.button`
  display: flex;
  font-family: Inter;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  width: 100px;
  padding: 10px 5px;
  border: none;
  background: ${COLORS.primary};
  border: 1px solid;
  border-color: ${COLORS.secondary};
  box-sizing: border-box;
  border-radius: 10px;
  color: ${COLORS.light};
  transition: 0.3s all;
  cursor: pointer;
  margin-left: 10px;

  &:hover {
    background: ${COLORS.error};
  }
`;

export const ActionArea = styled.div`
  display: flex;
  flex-direction: row;
  width: 50%;
`;
