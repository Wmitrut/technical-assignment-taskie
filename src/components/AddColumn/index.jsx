import { useRef, useState } from 'react';
import { ActionArea, AddButton, CancelButton, InputBoard, SaveButton } from './styles';

const AddColumn = (props) => {
  const [showNewColumnButton, setShowNewColumnButton] = useState(true);
  const inputBoard = useRef(null);
  const [value, setValue] = useState('');

  const handleInputFinish = () => {
    setShowNewColumnButton(true);
    addNewColumn(value);
    setValue('');
  };

  const handleCancel = () => {
    setShowNewColumnButton(true);
    setValue('');
  };

  function addNewColumn(title) {
    const newColumnOrder = Array.from(props.board.columnOrder);
    const newColumnId = 'column-' + Math.floor(Math.random() * 100000);
    newColumnOrder.push(newColumnId);

    const newColumn = {
      id: newColumnId,
      title: title,
      taskIds: [],
    };

    props.setBoard({
      ...props.board,
      columns: {
        ...props.board.columns,
        [newColumnId]: newColumn,
      },
      columnOrder: newColumnOrder,
    });
  }

  return (
    <>
      {showNewColumnButton ? (
        <AddButton
          type="button"
          onClick={() => {
            setShowNewColumnButton(false);
          }}>
          Add new Column
        </AddButton>
      ) : (
        <ActionArea>
          <InputBoard
            autoFocus
            type="text"
            placeholder="What is the title of your board?"
            value={value}
            onChange={(e) => setValue(e.target.value)}
            ref={inputBoard}
          />
          <CancelButton type="button" onClick={() => handleCancel()}>
            Cancel
          </CancelButton>
          <SaveButton type="button" onClick={() => handleInputFinish()} disabled={value === ''}>
            Save
          </SaveButton>
        </ActionArea>
      )}
    </>
  );
};

export default AddColumn;
