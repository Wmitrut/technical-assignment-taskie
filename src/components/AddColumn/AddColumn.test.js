import renderer from 'react-test-renderer';

import AddColumn from './index';

it('Renders correctly', () => {
  const tree = renderer.create(<AddColumn />).toJSON();
  expect(tree).toMatchSnapshot();
});
