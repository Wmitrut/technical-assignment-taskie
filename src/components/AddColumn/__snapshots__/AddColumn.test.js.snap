// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<button
  className="sc-bczRLJ bNiITr"
  onClick={[Function]}
  type="button"
>
  Add new Column
</button>
`;
